File: 이건희/level3/네트워크.py
def solution(n, computers):
    result = 0
    visited = []

    def check(x, visited):
        for i in range(n):
            if computers[x][i] == 1 and i not in visited:
                visited.append(i)
                check(i, visited)
                
    for i in range(n):
        if i not in visited:
            visited.append(i)
            result += 1
            check(i, visited)
    print(visited)        
    return result

File: 김찬우/프로그래머스 /Lv.3/네트워크.java
class Solution {
    
    static boolean visit[];
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        visit = new boolean[n];
        
        for(int i=0; i<n; i++){
            if(visit[i] == false){
                dfs(n,i,computers);
                answer++;
            }
            
        }
        
        return answer;
    }
    
    static void dfs(int n, int i, int[][] computers){
        
        visit[i] = true;
        
        for(int j=0; j<n; j++){
            
            if(i==j){
                continue;
            }
            
            if( visit[j] == false && computers[i][j]==1){
                dfs(n,j,computers);
            }
            
        }
        
        
    }
}
